2021-22/2021-12-06_70p

```diff
@@ -3,61 +3,58 @@
 from vexcode_vrc import *
 from math import sqrt
 
+def main():
+    drivetrain.set_drive_velocity(100,PERCENT)
+    fork_motor_group.spin_to_position(1800, DEGREES, wait=False)
+    goto( -920,  920, 0)
+    goto( -920,-1450, 0)
+    pick_up()
+    goto(  750,-1200, 0)
+    set_down()
+    goto( 1000,-1300, 1)
+    goto(  920, 1500, 0)
+    pick_up()
+    goto( -600, 1320, 0)
+    set_down()
+
+    goto(  600, 1320, 1)
+    goto(  600, -600, 1)
+    goto( 1400,-1350, 1)
+    goto( 1500, -900, 0)
+    pick_up()
+    goto( 1500,  100, 0)
+    set_down()
+    stop_project()
+
 def goto(target_x, target_y, reverse):
     x1 = gps.x_position(MM)
     y1 = gps.y_position(MM)
-    x2 = target_x
-    y2 = target_y
-    head = gps.heading()   # north is 0, but for the robot 0 is east - from start
-    delta_x = x2 - x1
-    delta_y = y2 - y1
+    delta_x = target_x - x1
+    delta_y = target_y - y1
     distance = math.sqrt(delta_x**2 + delta_y**2)     # pythagorean theorem
-    if ( delta_x ):
-        directio = - math.atan(delta_y / delta_x) * 180 / math.pi
+    if ( delta_x == 0 ):
+        if ( delta_y < 0):
+            direction = 90
+        else:
+            direction = 270
     else:
-        directio = 90
+        direction = - math.atan(delta_y / delta_x) * 180 / math.pi
     if ( delta_x < 0 ):
-        directio = directio + 180
-    monitor_variable("angle","x1","y1", "x2", "y2", "delta_x", "delta_y","head", "distance", "directio")
-    drivetrain.turn_to_heading(directio, DEGREES, wait=True)
-    if (reverse == 0):
-        drivetrain.drive_for(FORWARD, distance, MM, wait=True)
+        direction = direction + 180
+    if ( reverse != 0 ):
+        direction = direction + 180
+    if ( direction > 360 ):
+        direction = direction - 360
+    drivetrain.turn_to_heading(direction, DEGREES, wait=True)
+    if ( reverse != 0 ):
+        drivetrain.drive_for(REVERSE, distance, MM, wait=True)
     else:
-        drivetrain.drive_for(REVERSE, -distance, MM, wait=True)
+        drivetrain.drive_for(FORWARD, distance, MM, wait=True)
```

Comment:
When driving REVERSE the distance does not need to be inverted since the robot was already turned by 180 degrees (line 44)

Reply:
Thank you - I was wondering why it moved in the wrong direction